core/controllers/editor.py

```diff
@@ -775,3 +777,51 @@ def get(self, exploration_id):
         self.render_json({
             'unresolved_answers': unresolved_answers_with_frequency
         })
+
+
+class LearnerAnswerInfoHandler(EditorHandler):
+    """Handles the learner answer info for an exploration state."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_edit_exploration
+    def get(self, exploration_id):
+        """Handles the GET requests for for learner answer info for an
+        exploration state.
+        """
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.request.get('state_name')
+        if not state_name:
+            raise self.PageNotFoundException
+
+        entity_type = feconf.ENTITY_TYPE_EXPLORATION
+        state_reference = (
+            stats_models.LearnerAnswerDetailsModel
+            .get_state_reference_for_exploration(exploration_id, state_name))
+        learner_answer_details = stats_services.get_learner_answer_details(
+            entity_type, state_reference)
+        learner_answer_info_dict_list = [
+            learner_answer_info.to_dict() for learner_answer_info in
+            learner_answer_details.learner_answer_info_list]
+        self.render_json({
+            'learner_answer_info_dict_list': learner_answer_info_dict_list
+        })
+
+    @acl_decorators.can_edit_exploration
+    def delete(self, exploration_id):
+        """Deletes the learner answer info by the given id."""
+
+        if not constants.ENABLE_SOLICIT_ANSWER_DETAILS_FEATURE:
+            raise self.PageNotFoundException
+
+        state_name = self.request.get('state_name')
+        learner_answer_info_id = self.request.get('learner_answer_info_id')
```

Comment:
Can we use this in the url? like /handlereName/<exp-id>/<state-name>/<learner_answer_info_id>?

We cannot do this because the state name can be any Unicode value? or something else?

Reply:
@DubeySandeep can you explain about it? Thanks!

Reply:
He's asking if you can put the `learner_answer_info_id` also in the url.

Reply:
Actually if I do this then I have to make two different classes for get and delete function.

Reply:
This doesn't need to be in the URL, I think.

Reply:
Okay how should I do it?